inference/predictor_deployment.py

```diff
@@ -145,33 +146,71 @@ async def __call__(self, http_request: Request) -> Union[StreamingResponse, JSON
                 self.consume_streamer_async(streamer), status_code=200, media_type="text/plain"
             )
 
-    async def stream_response(self, prompt, config):
+    async def openai_call(self, prompt, config, streaming_response=True):
         prompts = []
         if isinstance(prompt, list):
-            if self.process_tool is not None:
-                prompt = self.process_tool.get_prompt(prompt)
-                prompts.append(prompt)
+            is_chat, is_prompts = get_input_format(prompt)
+            if is_chat:
+                if self.process_tool is not None:
+                    prompt = self.process_tool.get_prompt(prompt)
+                    prompts.append(prompt)
+                else:
+                    prompts.extend(prompt)
+            elif is_prompts:
+                yield HTTPException(
+                    400, "mulitple prompts is not supported now when using openai compatible api."
+                )
             else:
-                prompts.extend(prompt)
+                yield HTTPException(400, "invalid prompt format.")
         else:
             prompts.append(prompt)
```

Comment:
Need special care for vllm here, since there are two exec paths of vllm.generate, one for str, another for List[str].
The one for List[str] may not be efficient now. I will submit another PR to consolidate vllm.generate(str) and vllm.generate(List[str]). For now we need to pass str to vllm for single prompt (for benchmark).